noir-projects/noir-protocol-circuits/crates/private-kernel-lib/src/components/tail_output_composer/meter_gas_used.nr

```diff
@@ -34,6 +31,5 @@ pub fn meter_gas_used(data: CombinedAccumulatedData, gas_settings: GasSettings)
     metered_da_bytes += data.unencrypted_log_preimages_length as u32;
     metered_l2_gas += data.unencrypted_log_preimages_length as u32 * L2_GAS_PER_LOG_BYTE;
 
-    let teardown_gas = gas_settings.teardown_gas_limits;
-    Gas::new(metered_da_bytes * DA_GAS_PER_BYTE, metered_l2_gas) + Gas::tx_overhead() + teardown_gas
```

Comment:
Don't include `teardown_gas` for private-only tx.

(The users would have to set it to empty to avoid paying extra fees, which is quite annoying)